pghoard/postgres_command.py

```diff
@@ -45,10 +46,17 @@ def __init__(self, message, exit_code=EXIT_FAIL):
         self.exit_code = exit_code
 
 
-def http_request(host, port, method, path, headers=None):
+def http_request(host, port, method, path, headers=None, *, username=None, password=None):
     conn = HTTPConnection(host=host, port=port)
+    if headers is not None:
+        headers = headers.copy()
+    else:
+        headers = {}
+    if username is not None and password is not None:
+        auth_str = base64.b64encode(f"{username}:{password}".encode("utf-8")).decode()
```

Comment:
nit: should we check here username and password like
```
if username and password:
```
so we make sure they are not empty strings?
IIUC otherwise it will enable auth when empty string(s) are provided, I think at least username and ":" are required by RFC. But anyway in our Go implementation we require both to be not empty.

Reply:
Good call, adjusted